packages/grid/src/DeletableGridModel.ts

```diff
@@ -0,0 +1,42 @@
+import GridRange from './GridRange';
+import GridModel from './GridModel';
+import { AssertionError } from './errors';
+
+export function isDeletableGridModel(
+  model: GridModel
+): model is DeletableGridModel {
+  return (model as DeletableGridModel)?.isDeletable !== undefined;
+}
+
+export function assertIsDeletableGridModel(
+  model: GridModel
+): asserts model is DeletableGridModel {
+  if (!isDeletableGridModel(model)) {
+    throw new AssertionError(
+      `Expected 'model' to be deletable, but received ${model}`
+    );
+  }
+}
+
+/**
+ * Model for an deletable grid
+ */
+export interface DeletableGridModel extends GridModel {
+  isDeletable: boolean;
```

Comment:
We should still add a JS doc for this. It's possible to implement the method, but for that particular instance itself to not be deletable (e.g. `IrisGridTableModelTemplate` with an input table with no key columns).
```suggestion
  /** Whether this model is deletable or not */
  isDeletable: boolean;
```